codex-ui/src/vue/components/alert/useAlert.ts

```diff
@@ -0,0 +1,35 @@
+import { ref } from 'vue';
+import { createSharedComposable } from '@vueuse/core';
+import type { AlertInterface, AlertOptions, Alertype } from './Alert.types';
+
+/**
+ * Alert service hook
+ */
+export const useAlert = createSharedComposable((): AlertInterface => {
+  const alerts = ref<AlertOptions[]>([]);
+
+  /**
+   * Trigger alert component
+   * @param type type of alert (success, error, warning, info and default)
+   * @param opt alert options(timeout, message and icon)
+   */
+  function triggerAlert(type?: Alertype, opt?: Pick<AlertOptions, 'icon' | 'message' | 'timeout'>): void {
+    const alertIndex = alerts.value.findIndex((idx: AlertOptions) => idx.timeout === opt?.timeout);
```

Comment:
its not safe to use a timeout value as index, you can create a simple index explicitly.

Reply:
alerts.value.shift removes the oldest element from the array while inserting just 1, will only remove the latest element from the array